File: Python/gunting_batu_kertas.py
import os
import random

os.system("clear")

def gunting_batu_kertas():
    player_score = 0
    computer_score = 0
    screen_width = os.get_terminal_size().columns
    choices = ["Gunting", "Batu", "Kertas"]
    menu_choices = ("1. Gunting", "2. Batu", "3. Kertas", "4. Keluar")
    max_length = max(len(choice) for choice in menu_choices)
    
    while True:
        print("-" * screen_width)
        print(f"{'PERMAINAN GUNTING BATU KERTAS':^{screen_width}}\n")
        for choice in menu_choices:
            print(f"{choice:^{screen_width - max_length + len(choice)}}")
        print("\n" + "-" * screen_width)
        
        player_choices = int(input("\nSilahkan pilih (1, 2, 3, 4) : "))
        computer_choices = random.choice(choices)
        
        if player_choices == 1:
            player_choices = choices[0]
            if player_choices == choices[0] and computer_choices == choices[0]:
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Seri |-----':^{screen_width}}")
            elif player_choices == choices[0] and computer_choices == choices[1]:
                computer_score += 1
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Computer menang |-----':^{screen_width}}")
            elif player_choices == choices[0] and computer_choices == choices[2]:
                player_score += 1
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Player menang |-----':^{screen_width}}")
                
        elif player_choices == 2:
            player_choices = choices[1]
            if player_choices == choices[1] and computer_choices == choices[0]:
                player_score += 1
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Player menang |-----':^{screen_width}}")
            elif player_choices == choices[1] and computer_choices == choices[1]:
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Seri |-----':^{screen_width}}")
            elif player_choices == choices[1] and computer_choices == choices[2]:
                computer_score += 1
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Computer menang |-----':^{screen_width}}")
                
        elif player_choices == 3:
            player_choices = choices[2]
            if player_choices == choices[2] and computer_choices == choices[0]:
                computer_score += 1
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Computer menang |-----':^{screen_width}}")
            elif player_choices == choices[2] and computer_choices == choices[1]:
                player_score += 1
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Player menang |-----':^{screen_width}}")
            elif player_choices == choices[2] and computer_choices == choices[2]:
                print(f"\n\t      Player [{player_score}]:[{computer_score}] Computer")
                print(f"\n\tPlayer {'-->':^20} {player_choices}")
                print(f"\tComputer {'-->':^20} {computer_choices}")
                print(f"\n{'-----| Seri |-----':^{screen_width}}")
        
        elif player_choices == 4:
            print("")
            break
        else:
            print("\nPilihan yang tersedia hanya (1, 2, 3, dan 4)")
        
        repeat = input("\nIngin bermain lagi? (y/n) : ")
        
        if repeat.lower() == 'y':
            os.system("clear")
        elif repeat.lower() == 'n':
            print("")
            break
        else:
            print("\nInvalid Input!\n")
            break

if __name__ == "__main__":
    gunting_batu_kertas()

File: C++/gunting_batu_kertas.cpp
#include <iostream>
#include <cstdlib>

using namespace std;

int main() {
    while (true) {
        system("clear");
        
        cout << "--------------------------------------------------" << endl;
        cout << "          PERMAINAN GUNTING BATU KERTAS           " << endl << endl;
        cout << "                    1. Gunting                    " << endl;
        cout << "                    2. Batu                       " << endl;
        cout << "                    3. Kertas                     " << endl;
        cout << "                    \033[1;31m4. Keluar\033[0m                     " << endl << endl;
        cout << "--------------------------------------------------" << endl << endl;
        
        string choices[] = {"Gunting", "Batu", "Kertas"};
        int playerScore, computerScore;
        
        int playerChoices;
        cout << "Silahkan pilih (1, 2, 3, 4) : ";
        cin >> playerChoices;
        
        int computerChoices = rand() % 3; // range 0 to 2
        
        if (playerChoices == 1) {
            string player = choices[playerChoices - 1];
            string computer = choices[computerChoices];
            if (player == choices[0] && computer == choices[0]) {
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "                -----| Seri |-----" << endl << endl;
            } else if (player == choices[0] && computer == choices[1]) {
                computerScore++;
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "           -----| Computer menang |-----" << endl << endl;
            } else if (player == choices[0] && computer == choices[2]) {
                playerScore++;
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "            -----| Player menang |-----" << endl << endl;
            }
        } else if (playerChoices == 2) {
            string player = choices[playerChoices - 1];
            string computer = choices[computerChoices];
            if (player == choices[1] && computer == choices[0]) {
                playerScore++;
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "            -----| Player menang |-----" << endl << endl;
            } else if (player == choices[1] && computer == choices[1]) {
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "                -----| Seri |-----" << endl << endl;
            } else if (player == choices[1] && computer == choices[2]) {
                computerScore++;
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "           -----| Computer menang |-----" << endl << endl;
            }
        } else if (playerChoices == 3) {
            string player = choices[playerChoices - 1];
            string computer = choices[computerChoices];
            if (player == choices[2] && computer == choices[0]) {
                computerScore++;
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "           -----| Computer menang |-----" << endl << endl;
            } else if (player == choices[2] && computer == choices[1]) {
                playerScore++;
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "            -----| Player menang |-----" << endl << endl;
            } else if (player == choices[2] && computer == choices[2]) {
                cout << endl << "              Player [" << playerScore << "]:[" << computerScore << "] Computer" << endl << endl;
                cout << "        Player         -->          " << player << endl;
                cout << "      Computer         -->          " << computer << endl << endl;
                cout << "                -----| Seri |-----" << endl << endl;
            }
        } else if (playerChoices == 4) {
            cout << endl;
            break;
        } else {
            cout << endl << "Invalid Input!" << endl;
            break;
        }
        
        char repeat;
        cout << "Ingin bermain lagi? (y/n) : ";
        cin >> repeat;
        
        if (repeat == 'y') {
            // pass
        } else if (repeat == 'n') {
            cout << endl;
            break;
        } else {
            cout << endl << "Invalid Input!" << endl;
            break;
        }
    }
    
    return 0;
}